etc/make_small_lib.pl

```diff
@@ -0,0 +1,335 @@
+        $files .= ("\t" x $depth) . "\t\t>\r\n";
+        $files .= ("\t" x $depth) . "\t\t<Tool\r\n";
+        $files .= ("\t" x $depth) . "\t\t\tName=\"VCCLCompilerTool\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t\tAdditionalIncludeDirectories=\"\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t\tPreprocessorDefinitions=\"\"\r\n";
+        $files .= ("\t" x $depth) . "\t\t/>\r\n";
+        $files .= ("\t" x $depth) . "\t</FileConfiguration>\r\n";
+      }
+    }
+    $files .= ("\t" x $depth) . "</File>\r\n";
+  }
+  $files .= ("\t" x --$depth) . "</Filter>\r\n" for (@$last);
+  $files .= "\t</Files>";
+  return $files;
+}
+
+sub process_makefiles {
+  my ($path, @o) = @_;
+  my @all = map {$path.$sep . $_} @o;
+  my $var_o = prepare_makefile_variable("OBJECTS", @o);
+  (my $var_obj = $var_o) =~ s/\.o\b/.obj/sg;
+
+
+  # update MSVC project files
+  s/\.o/.c/ for @all;
+  my $msvc_files = prepare_msvc_files_xml(\@all, qr/NOT_USED_HERE/, ['Debug|Win32', 'Release|Win32', 'Debug|x64', 'Release|x64']);
+  for my $m (qw/libtommath_VS2008.vcproj/) {
+    my $old = read_file($path.$sep.$m);
+    my $new = $old;
+    $new =~ s|<Files>.*</Files>|$msvc_files|s;
+    if ($old ne $new) {
+      write_file($path.$sep.$m, $new);
+      warn "changed: $path$sep$m\n";
+    }
+  }
+
+  # update OBJECTS
+  for my $m (qw/ makefile makefile.shared makefile_include.mk makefile.msvc makefile.unix makefile.mingw /) {
+    my $old = read_file($path.$sep.$m);
+    my $new = $m eq 'makefile.msvc' ? patch_makefile($old, $var_obj)
+                                    : patch_makefile($old, $var_o);
+    if ($old ne $new) {
+      write_file($path.$sep.$m, $new);
+      warn "changed: $path$sep$m\n";
+    }
+  }
+}
+
+# TODO: that is not the best way to do it, but a simple one.
+sub write_header
+{
+  my ($path, @entries) = @_;
+  my $tcpath;
+  my $content = "/* LibTomMath, multiple-precision integer library -- Tom St Denis */\n";
+  $content = $content . "/* SPDX-License-Identifier: Unlicense */\n";
+  foreach my $entry (@entries) {
+    $entry =~ tr/[a-z]/[A-Z]/;
+    $entry = '#define BN_' . $entry . "_C\n";
+    $content = $content . $entry;
+  }
+  $tcpath = $path . $sep . 'tommath_class.h';
+  write_file($tcpath, $content);
+  warn "File $tcpath written\n";
+  $tcpath = $path . $sep . 'tommath_superclass.h';
+  write_file($tcpath, "");
+  warn "File $tcpath emptied\n";
+}
+
+sub gather_functions
+{
+  my $path = shift;
+  my %depmap;
+
+  if(-d $path) {
+    $path = $path . $sep . '*.c';
+  }
+
+  foreach my $filename (glob $path) {
+print "filename: " . $filename . " path: ". $path . "\n";
+    open(my $src, '<', $filename) or die "Can't open source file!\n";
+    read $src, my $content, -s $src;
+    close $src;
+
+    $filename = basename($filename);
+    $filename =~ s/^bn_|\.c$//g;
+
+    $content =~ s{/\*.*?\*/}{}gs;
+    my $list = "";
+    foreach my $line (split /\n/, $content) {
```

Comment:
does this already work together with bn_conversion bn_deprecated?

Reply:
`bn_deprecated`? It's deprecated, don't care.
`bn_conversion`? Not checked yet, but unlikely. If it gets splitted into single files it will work.

It also points out a disadvantage: this script relies on LTM's naming scheme for the source files instead of their content.
The naming scheme is needed because the content of some files get generated by a preprocessor macro. It's down to only one file now, `bn_conversion`. Please run `gcc -E` and C&P the generated code at the end into correctly named individual files. That would help a lot.

No, wait, there is also `bn_cutoffs.c`, which needs to be treated separately. But a single file with a handful of constants is acceptable.